app/assets/javascripts/application.js

```diff
@@ -91,8 +91,11 @@ function onContentLoad(){
 
   password_caps_lock_hint();
 
-  var tz = jstz.determine();
-  $.cookie('timezone', tz.name(), { path: '/', secure: location.protocol === 'https:' });
+  tfm.i18n.ready.then(() => {
+    var tz = jstz.determine();
+    $.cookie('timezone', tz.name(), { path: '/', secure: location.protocol === 'https:' });
```

Comment:
hey @tstrachota ... maybe you can use some `Cookies` with this PR? 🍪 

https://github.com/js-cookie/js-cookie

wdyt?

Reply:
maybe in the future but not now... i dunno...

Reply:
Yes @priley86 we should actually do that, thanks for the advice 👍 

I was also looking into that repository:
https://github.com/reactivestack/cookies

I think the `js-cookie` is better for us, for now, i still don't see us using `CookieProvider` or `withCookie` wrapper.

We actually trying to get ride of jquery which is a long process and migrating the cookies is a good step.

I open an issue and i hope to do it soon:
http://projects.theforeman.org/issues/22506

Reply:
I am fine w/ [react-cookie](https://github.com/reactivestack/cookies/tree/master/packages/react-cookie) too... and yes, very much prefer that API down the road if it becomes an option. Universal cookies are even more tasty! They could stream via `react-dom/server` within `apollo-server` in an isomorphic way, you know? Some kind of strange `renderToNodeStream` voodoo...haha

overall, i like the approach here w/ `react-intl` and utilizing the *componentRegistry*. I noticed Twitter doing something very similar [yesterday](http://nicolasgallagher.com/redux-modules-and-code-splitting/). I'm happy to move off of jquery as you have time...

Reply:
I'm ok with migrating from jquery cookies to some other solution in future, but it's out of the scope of this PR.